### PR TITLE
vault 1.5.0

### PR DESCRIPTION
**Pull request by ianferguson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Can it work without setting GOPATH in this version?

**Comment by ianferguson** [comment]
The GOPATH configuration appears to have been included since 2016 (1376f84db61)

The bump from 1.3.x to v1.4.0 (https://github.com/Homebrew/homebrew-core/pull/52725) added a GOPRIVATE var to work around a Vault issue, which was removed as intended in the v1.4.1 bump (https://github.com/Homebrew/homebrew-core/pull/54064/files), was that the removal you were referencing, or should the GOPATH configuration be removed as well, ideally?

**Comment by SMillerDev** [comment]
I meant the work being done here https://github.com/Homebrew/homebrew-core/issues/47627 to remove the use of GOPATH in favor of modules.

File: Formula/vault.rb [hunk 1]
```diff
@@ -23,20 +23,9 @@ class Vault < Formula
...
-    ENV.prepend_path "PATH", buildpath/"bin"
-
-    cd "src/github.com/hashicorp/vault" do
-      system "make", "bootstrap", "static-dist", "dev-ui"
-      bin.install "bin/vault"
-      prefix.install_metafiles
-    end
+    system "make", "bootstrap", "static-dist", "dev-ui"
+    bin.install "bin/vault"
+    prefix.install_metafiles
```
Comment (SMillerDev) [inline Formula/vault.rb:28]: This shouldn't be needed

Comment (ianferguson) [reply Formula/vault.rb]: removed in 36ba8b68f2

**Comment by ianferguson** [comment]
Looks like removing the GOPATH configuration results in this error:

**Comment by ianferguson** [comment]
Removing the GOPATH configuration resulted in this error in the CI tests:
```

Logs:
     /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/logs/vault/01.make
     /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/logs/vault/00.options.out
READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
vault 1.5.0 https://github.com/Homebrew/homebrew-core/pull/58416

==> brew audit vault --online --git --skip-style
==> brew uninstall --force gdbm go gox icu4c node node@10 openssl@1.1 pkg-config python@3.8 readline sqlite xz yarn
Error: 1 failed steps!
brew install --verbose --build-bottle vault
##[error]Process completed with exit code 1.
```

I can maybe look at this next week

**Comment by gromgit** [comment]
> Removing the GOPATH configuration resulted in this error in the CI tests:
> [...]

The actual error is [here](https://github.com/Homebrew/homebrew-core/pull/58416/checks?check_run_id=903483744#step:6:226):
```
--> Generating static assets
make: go-bindata-assetfs: No such file or directory
make: *** [static-assets] Error 1
```
`go-bindata-assetfs` seems to be one of the tools installed by `go get` during the build process, so you probably need to add something like the following at the start of the `install` method:
```
    ENV["GOBIN"] = buildpath/"bin"
    ENV.prepend_path "PATH", ENV["GOBIN"]
```

**Comment by chenrui333** [comment]
```
make: go-bindata-assetfs: No such file or directory
make: *** [static-assets] Error 1
```

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**